[ch133][ch137] As a user, I can see carousel and items of Quote box in About us screen

kimvcp opened the pull request:
https://app.clubhouse.io/cobda/story/133/as-a-user-i-can-see-quote-box-in-about-us-screen
https://app.clubhouse.io/cobda/story/137/as-a-user-i-can-see-quote-section-displaying-carousel-in-about-us-screen

## Why?

This **PR** is for adding Quote section under Goal section.

## Changes

- Add quote section containing `Carousel` and `Carousel Item`.
- `Carousel` is displaying 1 slide at a time.
- `Carousel Item` contains profile and quote box.
- Drag/Slide action is enabled.
- Indicators and buttons (`<`, `>`) are hidden for small screen, but auto playing is enabled instead.
- Locale supports both `EN` and `TH`.

## Screenshot(s)

**Before**

<img width="1149" alt="Screen Shot 2564-02-28 at 01 26 16" src="https://user-images.githubusercontent.com/32285706/109396406-f5f30600-7963-11eb-9abe-b16c19039f18.png">
 
**After**

Mobile

<img width="344" alt="Screen Shot 2564-02-28 at 03 40 54" src="https://user-images.githubusercontent.com/32285706/109399679-d6b1a400-7976-11eb-8f73-f71826f21a86.png">

Desktop

<img width="901" alt="Screen Shot 2564-02-28 at 03 41 05" src="https://user-images.githubusercontent.com/32285706/109399677-d31e1d00-7976-11eb-9e1c-8099eb88070d.png">

kimvcp committed:
[ch133][ch137] Init about us layout

kimvcp committed:
[ch133][ch137] Add simple item for carousel

kimvcp committed:
[ch133][ch137] Restructure files

kimvcp committed:
[ch133][ch137] Add background image for profile

kimvcp committed:
[ch133][ch137] Handle responsive profile card

kimvcp committed:
[ch133][ch137] Add double quotes image

kimvcp committed:
[ch133][ch137] Modify double quote image size

kimvcp committed:
[ch133][ch137] Apply responsiveness for quote box

kimvcp committed:
[ch133][ch137] Reduce sample words

kimvcp committed:
[ch133][ch137] Add header for mobile layout

kimvcp committed:
[ch133][ch137] Apply locale for th, en

kimvcp committed:
[ch133][ch137] Order css properties

kimvcp committed:
[ch133][ch137] Adjust padding and margin for all screen size

kimvcp committed:
[ch133][ch137] Add modifier for header

kimvcp committed:
[ch133][ch137] Handle carousel height of all screen

kimvcp committed:
[ch133][ch137] Resolve conflicts after rebase

kimvcp committed:
[ch133][ch137] Adjust size of quote box container

[ch133][ch137] Fix typo

[ch133][ch137] Refine code

[ch133][ch137] Adjust size of the container

[ch133][ch137] Reduce empty space for small screen

[ch133][ch137] Change colon image

Raksani commented on src/styles/components/_carousel.scss:
```diff
@@ -58,7 +82,7 @@
     border: none;
     background-color: $mine-shaft;
     color: $light;
```
Just a question. Should we set these buttons' outline as `none`? since the default is orange which is not going well with our signature color set 🎨
<img width="78" alt="Screenshot 2564-02-28 at 17 46 17" src="https://user-images.githubusercontent.com/32285705/109415716-dd7afe00-79ec-11eb-9808-dd012d0fa363.png">

kimvcp replied:
For your concern, how about using `outline: thin dotted;` instead. The main purpose is for [accessibility](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/What_is_accessibility)

Further reading [here](https://www.a11yproject.com/posts/2013-01-25-never-remove-css-outlines/)

Raksani commented on src/styles/components/_carousel-item.scss:
```diff
@@ -6,6 +6,23 @@
+    padding: rem(8px) 0 0 ;
+    box-shadow: 0 0 rem(20px) 0 box-shadow-opacity(0.16);
+
+    @include respond-above('sm') {
+      padding: rem(16px) 0;
+    }
```
I think `padding:0` looks good as well and not too far from the `title`. What do you think? 🤔 
<img width="544" alt="Screenshot 2564-02-28 at 17 52 19" src="https://user-images.githubusercontent.com/32285705/109415854-b53fcf00-79ed-11eb-8793-4eac02581235.png">

kimvcp replied:
The reason of my `padding` is to make sure that there is some space when `shadow` appeared. However, `shadow` seem to be hidden by a `Carousel`. It will surely be fixed so could I keep it there for now?

**Expected output**
<img width="602" alt="Screen Shot 2564-03-01 at 01 38 42" src="https://user-images.githubusercontent.com/32285706/109429564-dd035700-7a2e-11eb-9cfb-670eaf62df93.png">

Raksani commented on src/styles/components/_quote-box.scss:
```diff
@@ -0,0 +1,99 @@
+        image: url('/images/double-quote-left.png'),
+          url('/images/double-quote-left.png');
+        repeat: no-repeat;
+        size: rem(14px 12px), rem(14px 12px);
+        position: top left, bottom right;
+      }
+
+      @include respond-above('sm') {
+        background-size: rem(22px 20px), rem(22px 20px);
+      }
+
+      @include respond-above('md') {
+        background-size: rem(26px 24px), rem(26px 24px);
+      }
+
+      @include respond-above('lg') {
+        background-size: rem(30px 28px), rem(30px 28px);
+      }
+
+      @include respond-above('xl') {
+        background-size: rem(36px 32px), rem(36px 32px);
```
Actually, I tried `background-size: rem(x y)` and it applied for both components. 
```suggestion
        size: rem(14px 12px);
        position: top left, bottom right;
      }

      @include respond-above('sm') {
        background-size: rem(22px 20px);
      }

      @include respond-above('md') {
        background-size: rem(26px 24px);
      }

      @include respond-above('lg') {
        background-size: rem(30px 28px);
      }

      @include respond-above('xl') {
        background-size: rem(36px 32px);
```

kimvcp replied:
Good catch!

kimvcp committed:
[ch133][ch137] Resolve conflicts after rebase

kimvcp committed:
[ch133][ch137] Refine css for background image

[ch133][ch137] Refine css code

[ch133][ch137] Rename background image

LilBank commented on src/components/ProfileCard/index.tsx:
```diff
@@ -0,0 +1,26 @@
+  return (
+    <div className="profile-card profile-card--background-image">
+      <div className="profile-card__body">
+        <img src={imagePath} alt={alternate} className="profile-card__image" />
+        <div className="profile-card__label profile-card__label--name">
+          {name}
+        </div>
+        <div className="profile-card__label profile-card__label--role">
+          {role}
+        </div>
```
Try using `figure` and `figcaption` in this. See [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption)

LilBank commented on src/components/QuoteBox/index.tsx:
```diff
@@ -0,0 +1,23 @@
+      <header className="quote-box__header">
+        <h3 className="quote-box__title">{t('quoteSectionHeader')}</h3>
+      </header>
+      <div className="quote-box__container quote-box__container--background-image">
+        <p className="quote-box__content">{quote}</p>
+      </div>
```
`quote-box__container` inside `quote-box` just doesn't make sense when both of them are a block element.

LilBank commented on src/styles/components/_about-us-section.scss:
```diff
@@ -1,6 +1,12 @@
   box-sizing: border-box;
-  padding: rem(16px) 0;
+
+  @include respond-above('sm') {
+    padding: rem(32px) 0;
+  }
```
Screen size within `sm` - `md` shouldn't have bigger `paddings` than `< sm`. Please use the same values from `home-section`.

LilBank commented on src/styles/components/_profile-card.scss:
```diff
@@ -0,0 +1,109 @@
+
+  &--background-image {
+    background: {
+      image: url('/images/blue-ellipse.png');
```
Please rename the file to something like `profile-card-background` or any other meaningful names.

LilBank commented on src/styles/components/_profile-card.scss:
```diff
@@ -0,0 +1,109 @@
+      }
+    }
+
+    &--name {
```
How about renaming this modifier to `large` and `small` for the other one?

kimvcp replied:
Since we only have 2 modifiers for this label. How about `primary` and `secondary` ?

LilBank commented on src/styles/components/_profile-card.scss:
```diff
@@ -0,0 +1,109 @@
+    flex-direction: column;
+    box-sizing: border-box;
+    box-shadow: 0 0 rem(20px) 0 box-shadow-opacity(0.16);
+    width: fit-content;
```
Don't see any effect of this property, please recheck.

LilBank commented on src/styles/components/_quote-box.scss:
```diff
@@ -0,0 +1,99 @@
+
+.quote-box {
+  width: 100%;
+  height: fit-content;
```
Don't see effect of this property, please recheck.

LilBank commented on src/styles/components/_quote-box.scss:
```diff
@@ -0,0 +1,99 @@
+  height: fit-content;
+
+  @include respond-above('md') {
+    margin-top: rem(24px);
```
How about using `margin-top` instead?

kimvcp replied:
I think you mean `padding-top` instead

kimvcp committed:
[ch133][ch137] Adjust size of container and element

[ch133][ch137] Adjust carousel height

[ch133][ch137] Adjust size without natural height

kimvcp committed:
[ch133][ch137] Restructure component

kimvcp committed:
[ch133][ch137] Modify carousel layout

kimvcp committed:
[ch133][ch137] Resolve conflicts after rebase

Raksani commented on src/styles/components/_about-us-section.scss:
```diff
@@ -13,4 +15,49 @@
+
+  &--light {
+    height: rem(200px);
+    background-color: teal;
```
Please add `teal` color as a variable. And according to [the way to order CSS properties](https://css-tricks.com/poll-results-how-do-you-order-your-css-properties/), we have to separate `color` ☺️

Raksani commented on src/styles/components/_carousel.scss:
```diff
@@ -5,35 +5,63 @@
+      }
+
+      @include respond-above('lg') {
+        max-width: 93.5vw;
```
I'm just curious, why we use `vw` unit?

kimvcp replied:
The carousel's width from lib `pure-react-carousel` does not shrink along with the screen. `vw` is set to ensure that it will shrink to the screen properly.
PS. You can try this yourself, setting `max-width` to something larger and `min-width` accordingly

kimvcp committed:
[ch133][ch137] Remove unused code

LilBank commented on src/components/ProfileCard/index.tsx:
```diff
@@ -0,0 +1,26 @@
+  return (
+    <div className="profile-card profile-card--background-image">
+      <div className="profile-card__body">
+        <img src={imagePath} alt={alternate} className="profile-card__image" />
```
Currently the `margin` is applied on the `img`, would be better to have an `image-container`.

kimvcp replied:
Sure, I would go with `image-wrapper` instead since `container` means to have many elements inside. In this case, we only have one. What do you think?

Further reading [here](https://css-tricks.com/best-way-implement-wrapper-css/)

LilBank commented on src/components/QuoteBoxCarousel/Item.tsx:
```diff
@@ -0,0 +1,27 @@
+  readonly quote: string
+}
+
+const CarouselItem = ({ imagePath, alternate, name, role, quote }: CarouselItem) => {
```
Currently you are using arguments from this function to create `ProfileCard` and `QuoteBox`, what do you think of changing `CarouselItem` parameters to accept `ProfileCard` and `QuoteBox` instead?

kimvcp replied:
Very interesting 👏

LilBank commented on src/components/QuoteBoxCarousel/index.tsx:
```diff
@@ -0,0 +1,68 @@
+    alternate: 'Adidas fashion sportswear',
+    name: t('name'),
+    role: t('role'),
+    quote: t('quoteSectionContent'),
```
```suggestion
    quote: t('quoteSectionContent')
```

LilBank commented on src/styles/components/_about-us-section.scss:
```diff
@@ -13,4 +15,51 @@
+    margin: 0;
+  }
+
+  &__light {
+    background-color: $light;
+  }
```
Please recheck the usage of this class.

kimvcp committed:
[ch133][ch137] Refine code for more understandability
